lib/active_merchant/billing/gateways/cybersource_rest.rb

```diff
@@ -0,0 +1,104 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class CybersourceRestGateway < Gateway
+      self.test_url = 'https://apitest.cybersource.com'
+      self.live_url = 'https://api.cybersource.com'
+
+      self.supported_countries = ['US']
+      self.default_currency = 'USD'
+      self.supported_cardtypes = %i[visa master american_express discover diners_club jcb maestro elo union_pay cartes_bancaires mada]
+
+      self.homepage_url = 'http://www.cybersource.com'
+      self.display_name = 'CyberSourceRest'
+
+      def initialize(options = {})
+        requires!(options, :merchant_id, :public_key, :private_key)
+        super
+      end
+
+      def purchase(money, payment, options = {})
+        commit('sale', post)
+      end
+
+      def authorize(money, payment, options = {})
+        commit('authonly', post)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript
+      end
+
+      private
+
+      def commit(action, parameters)
+        url = (test? ? test_url : live_url)
+        response = parse(ssl_post(url, post_data(action, parameters)))
+
+        Response.new(
+          success_from(response),
+          message_from(response),
+          response,
+          authorization: authorization_from(response),
+          avs_result: AVSResult.new(code: response['some_avs_response_key']),
+          cvv_result: CVVResult.new(response['some_cvv_response_key']),
+          test: test?,
+          error_code: error_code_from(response)
+        )
+      end
+
+      # This implementation follows the detailed Cybersource guide on how create the
+      # request signature, see: https://developer.cybersource.com/docs/cybs/en-us/payments/developer/all/rest/payments/GenerateHeader/httpSignatureAuthentication.html
+      def get_http_signature(resource, digest, http_method = 'post', gmtdatetime = Time.now.httpdate)
+        string_to_sign = {
+          host: URI.parse(test? ? test_url : live_url).host,
+          date: gmtdatetime,
```

Comment:
we can use directly the Time.now.httpdate here instead, what happen if the date format put int the param is wrong ❓

Reply:
@javierpedrozaing nop, because the date should be the same for the request http header and the one used in the signature, in fact, the most reasonable change is to remove the default value from the method signature.